src/ui/Island/Cursor/index.tsx

```diff
@@ -0,0 +1,83 @@
+import React from "react"
+import CursorLabel, { CursorLabelProps } from "./CursorLabel"
+
+export default function IslandCursor({
+  cursor,
+  userInfo,
+  extraCursor,
+}: CursorLabelProps) {
+  if (!cursor) return null
```

Comment:
you don't need this additional check when you check it during mapping in src/ui/Island/IslandPresence.tsx and you also set it required in type.